internal/sink/sink.go

```diff
@@ -431,16 +429,18 @@ func (s *Sink) Run() error {
 	s.namespaceRefetch = time.AfterFunc(s.config.NamespaceRefetch, refetch)
 
 	// Reset state
-	s.running = true
+	s.isRunning.Store(true)
+	defer s.Close()
```

Comment:
`sink.Close()` is called in main by the group function as well (which makes sense IMO). Why is it called here?

Reply:
My thinking was to make sure that we properly "cleanup" if `sink.Run()` returns due to ctx cancellation, but I agree that this is not the best of handling this and it is better tor have the grpup function to call the `sinkClose()` when `sink.Run()` returns. I'll push an update ti fix this.